automation_tools/__init__.py

```diff
@@ -1622,6 +1623,30 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
+def host_pings(host):
+    """This ensures the given IP/hostname pings succesfully.
+
+    :param host: A string. The IP or hostname of host.
+
+    """
+    if host is None:
+        print("The hostname/IP is not provided to ping. \
+        Terminating the execution !!")
+        sys.exit(1)
+    while True:
+        command = subprocess.Popen(
+            'ping -c1 {0}'.format(host),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True
+        )
+        output = command.communicate()[0]
+        if '1 received' in output:
```

Comment:
You're currently looping and using `-c 1`, so you can simply check for a return code of 0.

```
$ ping -c 1 192.168.0.1 >/dev/null; echo $?
0
```

But yes, I do think that it is better to inspect return codes than prose. Prose can change at any time. There's absolutely no guarantee of stability.